presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/QueryBuilder.java

```diff
@@ -198,7 +211,109 @@ else if (typeAndValue.getType() instanceof CharType) {
                 throw new UnsupportedOperationException("Can't handle type: " + typeAndValue.getType());
             }
         }
+        log.debug("Query is not Pushed down ");
+        log.debug("Actual Query :: ", sql.toString());
+        return statement;
+    }
+
+    private boolean checkIfJoinQuery(Optional<List<ConnectorTableHandle>> joinPushdownTables)
+    {
+        return (joinPushdownTables != null && joinPushdownTables.isPresent() && !joinPushdownTables.get().isEmpty());
+    }
+
+    private PreparedStatement buildJoinSql(
```

Comment:
Instead of completely forking the buildSql code (and repeating its parts), please refactor by creating smaller methods that take in a input of`StringBuilder sql` and append SQL that is needed to query the joined tables using the `FROM table1, table2,.., tableX` format. This will make writing tests for buildSQL much easier too